include/h2o.h

```diff
@@ -337,6 +341,12 @@ typedef enum h2o_proxy_zerocopy_mode {
     H2O_PROXY_ZEROCOPY_ALWAYS
 } h2o_proxy_zerocopy_mode_t;
 
+typedef enum h2o_proxy_expect_mode {
+    H2O_PROXY_EXPECT_DISABLED,
+    H2O_PROXY_EXPECT_ENABLED,
+    H2O_PROXY_EXPECT_FORWARD,
```

Comment:
Can we have doc-comments explaining how these work? I see the explanation on the PR, but people reading code do not necessarily go back to the original PR (and the PR never gets updated as we change the definition).

Reply:
Addressed in https://github.com/h2o/h2o/pull/3419/commits/71e29f72f07d4dfdcff1adbb073e33c3bdf19335